rules/tree.go

```diff
@@ -47,33 +47,43 @@ type Tree[T1 any, T2 any] struct {
 // If the result matches one of the node values on the next level, we move to that node, otherwise we exit.
 // If a leaf node is reached, it's result functions are executed on the provided result payload.
 func (t *Tree[T1, T2]) Run(payload *T1, result *T2) error {
-	var matchingValue string
-	currNode := t.Root
+	if t.Root == nil {
+		return errors.New("tree root is nil")
+	}
+
+	if len(t.Root.Children) == 0 {
+		return errors.New("At least one child is required so a schema function value can match against it")
+	}
```

Comment:
Check if this is necessary after reverting those other couple changes.

Reply:
It isn't. Both checks at the beginning turned out to be unnecessary but, the downside is that we don't throw an error when `t.Root` is `nil` or `t.Root.Children` is `nil` or empty. Let me know if you agree with the latest approach.